crates/indexer/src/helper.rs

```diff
@@ -117,18 +83,47 @@ pub enum GwLog {
     },
 }
 
-fn parse_sudt_log_data(data: &[u8]) -> ([u8; 20], [u8; 20], u128) {
-    assert_eq!(data[0], 20);
-    let mut from_address = [0u8; 20];
-    from_address.copy_from_slice(&data[1..21]);
+fn parse_sudt_log_data(data: &[u8]) -> anyhow::Result<(RegistryAddress, RegistryAddress, u128)> {
+    let mut start = 0;
+    let mut end = 0;
 
-    let mut to_address = [0u8; 20];
-    to_address.copy_from_slice(&data[21..41]);
+    end += {
+        let from_address_byte_size = u32::from_le_bytes(data[4..8].try_into()?);
+        if from_address_byte_size == 0 {
+            8
+        } else {
+            28
```

Comment:
I am confused about this assignment.
Would it be possible to add some comments to the code: what's our expected format of `data`?

Reply:
Add some comments of this function